app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -39,17 +42,37 @@ class AllFilesFragment : Fragment() {
     @RequiresApi(Build.VERSION_CODES.R)
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        val items = mutableListOf<ListModel>()
-        val path = Environment.getRootDirectory().absolutePath
-        File(path).walk().forEach {
-            items.add(ListModel(it.name))
-        }
 
-        adapter = ListAdapter()
         val layoutManager = LinearLayoutManager(context)
+        adapter = ListAdapter(requireContext(), fetchList(), this)
```

Comment:
In your next Pull Request please create a ViewModel and use it for fetchng the file list. We can discuss more about the MVVM architecture in a call. Here's a link for reference about how ViewModel's work https://developer.android.com/topic/libraries/architecture/viewmodel .